src/database/database-postgresql.cpp

```diff
@@ -92,11 +92,18 @@ void Database_PostgreSQL::connectToDatabase()
 
 void Database_PostgreSQL::verifyDatabase()
 {
-	if (PQstatus(m_conn) == CONNECTION_OK)
-		return;
+	// Verify DB connection with ping
+	try {
+		ping();
+	} catch (const DatabaseException &e) {
+		// If ping failed, show the error and try reconnect
+
+		errorstream << e.what() << std::endl;
+		PQreset(m_conn);
 
-	PQreset(m_conn);
-	ping();
+		actionstream << "Reconnecting to database " << m_connect_string << std::endl;
```

Comment:
I think this is better merged into a single error message.

Reply:
fine